### PR TITLE
Introduce thermal containers

### LINKED ISSUE #134: Suggestion: Distinct access to thermal models
As of now, thermal models can only be accessed via electrical models, that do have a reference to those. However, a `ThermalBus` or a `ThermalStorage` e.g. could connect several thermal models (and thereby electrical ones as well). From my point of view it is an in-homogeneity with regard to electrical models.

I would suggest two options:
*Option 1*

- Have a set of `ThermalBusInput` in `RawGridElements`
- Have a set of `CylindricalStorageInput`, `ThermalHouseInput` in `SystemParticipants`

*Option 2*
Create a new `ThermalContainer` and assign it to a `GridContainer`

### PR DESCRIPTION
**Pull request by ckittl**
Resolves #134

### CODE REVIEW

**Comment by ckittl** [comment]
Sorry for not passing the quality gate. But I won't write tests for records. ;-)

File: src/main/java/edu/ie3/datamodel/models/input/container/ThermalGrid.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import edu.ie3.datamodel.models.input.thermal.ThermalStorageInput;
+import java.util.*;
+
+/**
+ * Container object to denote a fully connected thermal "grid". As there are currently no branch
+ * elements, a grid always only consists of one {@link ThermalBusInput} and all its connected {@link
+ * edu.ie3.datamodel.models.input.thermal.ThermalUnitInput}s
+ */
+public record ThermalGrid(
+    ThermalBusInput bus, Set<ThermalHouseInput> houses, Set<ThermalStorageInput> storages)
```
Comment (sebastian-peter) [inline src/main/java/edu/ie3/datamodel/models/input/container/ThermalGrid.java:20]: Do you think it'd pay off to split thermal storages into their subtypes CylindricalStorage, ...?

File: src/main/java/edu/ie3/datamodel/models/input/container/ThermalUnits.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.*;
+
+public record ThermalUnits(Set<ThermalHouseInput> houses, Set<ThermalStorageInput> storages)
+    implements InputContainer<ThermalUnitInput> {
+  /**
+   * Container to group together all {@link ThermalUnitInput}s
+   *
+   * @param houses Available houses
+   * @param storages Available storage
+   */
```
Comment (sebastian-peter) [inline src/main/java/edu/ie3/datamodel/models/input/container/ThermalUnits.java:20]: Doesn't this ScalaDoc fit the class better, at the top?

File: src/main/java/edu/ie3/datamodel/models/input/thermal/ThermalUnitInput.java [hunk 3]
```diff
@@ -7,20 +7,19 @@
...
-public abstract class ThermalUnitInput extends AssetInput implements HasThermalBus {
+public abstract class ThermalUnitInput extends ThermalInput implements HasThermalBus {
   /** The thermal bus, a thermal unit is connected to. */
   private final ThermalBusInput thermalBus;
 
   /**
    * @param uuid Unique identifier of a certain thermal input
    * @param id Identifier of the thermal unit
-   * @param thermalBus hermal bus, a thermal unit is connected to
+   * @param thermalBus thermal bus, a thermal unit is connected to
```
Comment (sebastian-peter) [inline src/main/java/edu/ie3/datamodel/models/input/thermal/ThermalUnitInput.java:22]: hank you very much :)

**Review by sebastian-peter — CHANGES_REQUESTED** [review]
I agree testing e.g. the getters of a record does not make sense, but testing `allEntitiesAsList()` might? Thanks anyway for the work.

File: src/main/java/edu/ie3/datamodel/models/input/thermal/ThermalUnitInput.java (hunk 3 above)
Comment (ckittl) [reply src/main/java/edu/ie3/datamodel/models/input/thermal/ThermalUnitInput.java:22]: ... or traveling with deutsche bahn!

**Review by sebastian-peter — APPROVED** [review]
Ok I just provided some really quick tests for `allEntitiesAsList()` and now the test coverage is good :)
Thanks for the contribution!

**Pull request merged**